### DISCUSSION TITLE
How to use the add/delete passcode functionality?

### DISCUSSION THREAD

**Discussion by purugupta99** [jbergler/hass-ttlock#93] [Q&A] [answered]
## Describe the bug

I am trying to access the add/delete passcode functionality via the ttlock device dashboard, such that I can set up the passcodes using the HA UI but I didn't see it over there to update/delete the passcodes. Can you help me with how I can use this service to update the passcodes pls? (I'm using the latest release version v0.6.1)

### To Reproduce

1. Go to TTLock dashboard from integrations
2. Click on the devices link to view all the devices under the integration
3. Select a device from the `/config/devices/dashboard` page
4. Under the controls and sensor widget, the service is not available

### Expected behavior

The service to add new passcodes should have been available under the controls settings

## Diagnostic info
[config_entry-ttlock-3c54564471c861d73666e707868b832a.json.txt](https://github.com/jbergler/hass-ttlock/files/13920660/config_entry-ttlock-3c54564471c861d73666e707868b832a.json.txt)

**Comment by purugupta99** [comment]
@PedroDelCargo, can you help here please? (I saw that you helped with the initial PR)

**Comment by PedroDelCargo** [comment, marked as answer]
> @PedroDelCargo, can you help here please? (I saw that you helped with the initial PR)

Hi,
There is no UI for this. It is a service that can be called with automations, scripts or from Development tools -> Services.
The service needs parameters to be set when you use it (Lock id, passcode, passcode name, start date/time and end date/time).
Personally, I use it to automate passcodes creation/deletion. I created automations that work fine. If you need help to create such an automation, let me know.

**Reply by wyliletv** [reply]
Could you post an example please?

**Reply by jbergler** [reply]
Sure! The schema is exposed through developer tools (which is how Home Assistant recommends services are documented)
If you don't see the UI when you select the service there should be a "Go to UI Mode" or "Go to YAML Mode" button to flip between the two.

![Screenshot from 2024-02-15 09-02-21](https://github.com/jbergler/hass-ttlock/assets/698597/56586825-3b20-46a2-9d35-7ae81d26224e)

There are also the docs on how to make service calls which might help with broader context.
https://www.home-assistant.io/docs/scripts/service-calls/

**Comment by mremedi2023** [comment]
Hi everyone, first of all congratulations on the integration, it works very well. Does anyone know how to delete a keyboard from HA? I deleted it from the application because I put a new one but it still appears linked and I can't delete it. Thanks

**Reply by PedroDelCargo** [reply]
Hi,
Did you try to reload the integration?

**Comment by mremedi2023** [comment]
Yes I did, but I still have 2 keyboards in the app and 3 in HA

**Reply by PedroDelCargo** [reply]
You can try to remove manually each entity from the keyboard you don't need and if the device is still there and you really want to remove it, I'm afaraid you have to uninstall and reinstall the integration...

**Reply by jbergler** [reply]
I should probably have a better answer to this, but where are the keyboards showing up?
Can you post some screenshots?

**Comment by mremedi2023** [comment]
Fixed, I uninstalled the integration and reinstalled it and now I only see 2 keyboards. Thank you very much
